importer_client/python/timesketch_import_client/importer.py

```diff
@@ -708,21 +708,24 @@ def celery_task_id(self):
         """Return the celery task identification for the upload."""
         return self._celery_task_id
 
-    def close(self):
-        """Close the streamer."""
+    def close(self, analyzer_names=None):
+        """Close the streamer, and run the analyzers based on the analyzer_names"""
         try:
             self._ready()
         except ValueError:
             return
 
+        if analyzer_names is None:
+            analyzer_names = []
```

Comment:
Instead of this check here you could also set the default for the parameter directly to `[]`.

Reply:
Thanks! Originally that was this implementation (set the default for the parameter directly to `[]`), but there is a pylint warning saying that defaulting parameter to `[]` is dangerous.

Reply:
Thanks! Originally that was this implementation (set the default for the parameter directly to []), but there is a pylint warning saying that defaulting parameter to [] is dangerous.